From 52f84e1c3eb060fbddd1636e3355c10aa49891be Mon Sep 17 00:00:00 2001
From: Wu Zhangjin <falcon@tinylab.org>
Date: Sat, 13 Jun 2020 02:31:50 +0800
Subject: [PATCH 11/12] gcc-9: fix up conflicts between -mloongson-mmi and
 -msoft-float

loongson-mmi must be disabled for soft float is used in kernel.

ref: http://www.loongnix.org/index.php/User_manual

Signed-off-by: Wu Zhangjin <falcon@tinylab.org>
---
 arch/mips/Makefile | 1 +
 1 file changed, 1 insertion(+)

diff --git a/arch/mips/Makefile b/arch/mips/Makefile
index 98e78bf..c995139 100644
--- a/arch/mips/Makefile
+++ b/arch/mips/Makefile
@@ -89,6 +89,7 @@ all-$(CONFIG_SYS_SUPPORTS_ZBOOT)+= vmlinuz
 #
 cflags-y			+= -G 0 -mno-abicalls -fno-pic -pipe
 cflags-y			+= -msoft-float
+cflags-$(CONFIG_CPU_LOONGSON3)  += $(call cc-option, -mno-loongson-mmi)
 LDFLAGS_vmlinux			+= -G 0 -static -n -nostdlib
 KBUILD_AFLAGS_MODULE		+= -mlong-calls
 KBUILD_CFLAGS_MODULE		+= -mlong-calls
-- 
2.7.4

